src/jit/morph.cpp

```diff
@@ -19217,6 +19217,32 @@ Compiler::fgWalkResult Compiler::fgMarkAddrTakenLocalsPreCB(GenTree** pTree, fgW
         // them as AXC_IndWide.
         //
 
+        case GT_CALL:
+        {
+            // Scan for byref args
+            GenTreeCall* const call = tree->AsCall();
+
+            if (call->gtCallObjp)
```

Comment:
Isn't this a bit too conservative? Surely people won't call a struct method and then expect that they can access memory past `this`?

Reply:
Yeah, that probably is too conservative.